src/protocolgame.cpp

```diff
@@ -516,20 +516,51 @@ void ProtocolGame::GetTileDescription(const Tile* tile, NetworkMessage& msg)
 		}
 	}
 
-	const CreatureVector* creatures = tile->getCreatures();
-	if (creatures) {
-		for (const Creature* creature : boost::adaptors::reverse(*creatures)) {
-			if (!player->canSeeCreature(creature)) {
-				continue;
+	if (!loggedIn && tile->getPosition() == player->getPosition()) {
+		bool playerSpawned = false;
+		const CreatureVector *creatures = tile->getCreatures();
+		if (creatures) {
+			for (const Creature *creature : boost::adaptors::reverse(*creatures)) {
+				if (!player->canSeeCreature(creature)) {
+					continue;
+				}
+
+				if (creature == player) {
+					playerSpawned = true;
+				}
+
+				bool known;
+				uint32_t removedKnown;
+				checkCreatureAsKnown(creature->getID(), known, removedKnown);
+				AddCreature(msg, creature, known, removedKnown);
+
+				if (count == 8 && playerSpawned == false) { // player still not spawned and we need to send him too
```

Comment:
Can count be equal to 9 at the first iteration?

Reply:
Technically it could if there are 8 tile items on the ground and the ground. However, the tile "items" that I checked don't quite stack that much in any practical way - you can probably only stack a few types of items on each other (ground, second ground type for half-tile, grass, different grass come up to 4 which is max what I found out).

Creating 8 "counters" (like in the shop) makes your character disappear if you `/a 1` onto it in game and the client does not crash. However, when you try to log in to such position, the server would put you around it as the tile is not walkable. Therefore, I haven't found a way to stack that many walkable tile items to go over 8.

Now, if we do go over 8, then the character won't be sent to the client and it will MOST LIKELY crash due to the same issue this PR is supposed to fix. If we change this line to `count >= 8`, it will probably still crash due to too many things on a tile.

This PR really addresses the issue of multiple creatures on one tile, which seems to be the most probably reason for the current issue of client crash.

Thoughts @djarek ?